contracts/ShortNameClaims.sol

```diff
@@ -82,80 +106,193 @@ contract ShortNameClaims is Ownable {
     }
 
     /**
-     * @dev Submits a claim for an exact match (eg, fooeth.test -> foo.eth).
+     * @dev Submits a claim for prefix match (eg, fooeth.test -> foo.eth).
      *      Claimants must provide an amount of ether equal to 365 days'
      *      registration cost; call `getClaimCost` to determine this amount.
      *      Claimants should supply a little extra in case of variation in price;
      *      any excess will be returned to the sender.
      * @param name The DNS-encoded name of the domain being used to support the
      *             claim.
      * @param claimant The address of the claimant.
+     * @param email An email address for correspondence regarding the claim.
      */
-    function submitPrefixClaim(bytes memory name, address claimant) public payable {
+    function submitPrefixClaim(bytes memory name, address claimant, string memory email) public payable {
         bytes memory firstLabel = bytes(getLabel(name, 0));
         require(firstLabel.equals(firstLabel.length - 3, bytes("eth")));
-        handleClaim(string(firstLabel.substring(0, firstLabel.length - 3)), name, claimant);
+        handleClaim(string(firstLabel.substring(0, firstLabel.length - 3)), name, claimant, email);
+    }
+
+    /**
+     * @dev Closes the claim submission period.
+     *      Callable only by the owner.
+     */
+    function closeClaims() external onlyOwner inPhase(Phase.OPEN) {
+        phase = Phase.REVIEW;
+    }
+
+    /**
+     * @dev Ratifies the current set of claims.
+     *      Ratification freezes the claims and their resolutions, and permits
+     *      them to be acted on.
+     */
+    function ratifyClaims() external inPhase(Phase.REVIEW) {
+        // Only the ratifier can ratify.
+        require(msg.sender == ratifier);
+        // Can't ratify until all claims have a resolution.
+        require(pendingClaims == 0);
+        phase = Phase.FINAL;
     }
 
-    function approveClaim(bytes32 claimId) onlyOwner public {
+    /**
+     * @dev Cleans up the contract, after all claims are resolved.
+     *      Callable only by the owner, and only in final state.
+     */
+    function destroy() external onlyOwner inPhase(Phase.FINAL) {
+        require(unresolvedClaims == 0);
+        selfdestruct(toPayable(ratifier));
+    }
+
+    /**
+     * @dev Sets the status of a claim to either APPROVED or DECLINED.
+     *      Callable only during the review phase, and only by the owner or
+     *      ratifier.
+     * @param claimId The claim to set the status of.
+     * @param status The status to set - must be APPROVED or DECLINED.
+     */
+    function setClaimStatus(bytes32 claimId, Status status) public inPhase(Phase.REVIEW) {
+        // Only callable by owner or ratifier
+        require(msg.sender == owner() || msg.sender == ratifier);
+
+        // Can't set claim back to pending or to withdrawn
+        require(status == Status.APPROVED || status == Status.DECLINED);
```

Comment:
If you change from claim.status to claim.approved(yes/no) you don't need this check